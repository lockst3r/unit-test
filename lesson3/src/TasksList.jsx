import React, {Component} from 'react';
import Task from './Task';
import PropTypes from 'prop-types';

const TasksList = ({ tasks, handleTaskStatusChange, handleTaskDelete }) => {

    const sortedList = tasks
        .slice() 
        .sort((a, b) => a.done - b.done);
    return(
        
            <ul className="list">
                    {sortedList.map(task => {
                    return(
                    <Task key={task.id} {...task}
                        onDelete={handleTaskDelete} 
                        onChange={handleTaskStatusChange}
                    />
                )})}
            </ul>
    );

};

TasksList.propTypes = {
    tasks: PropTypes.arrayOf(
        PropTypes.shape({
            text: PropTypes.string,
            done: PropTypes.bool,
            id: PropTypes.string,
        }),
    ).isRequired,
    handleTaskStatusChange: PropTypes.func.isRequired,
    handleTaskDelete: PropTypes.func.isRequired,
};

export default TasksList;