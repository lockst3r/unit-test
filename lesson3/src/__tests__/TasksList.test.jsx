import React from "react";
import { shallow } from "enzyme";
import TasksList from "../TasksList";

describe("<TasksList />", () => {
  it('Should display tasks list sorted by "done" status ', () => {
    const props = {
      handleTaskStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
      tasks: [
        {
          text: "Task 1",
          done: true,
          id: "task-id-1",
        },
        {
          text: "Task 2",
          done: false,
          id: "task-id-2",
        },
        {
            text: "Task 3",
            done: false,
            id: "task-id-3",
          },
      ],
    };
    const wrapperComponent = shallow(<TasksList {...props} />);
    expect(wrapperComponent).toMatchSnapshot();
  });
});
