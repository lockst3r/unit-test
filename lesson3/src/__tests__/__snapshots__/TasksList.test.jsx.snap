// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> Should display tasks list sorted by "done" status  1`] = `
<ul
  className="list"
>
  <Task
    done={false}
    id="task-id-2"
    key="task-id-2"
    onChange={[MockFunction]}
    onDelete={[MockFunction]}
    text="Task 2"
  />
  <Task
    done={false}
    id="task-id-3"
    key="task-id-3"
    onChange={[MockFunction]}
    onDelete={[MockFunction]}
    text="Task 3"
  />
  <Task
    done={true}
    id="task-id-1"
    key="task-id-1"
    onChange={[MockFunction]}
    onDelete={[MockFunction]}
    text="Task 1"
  />
</ul>
`;
